### PR TITLE
Tutorial - Basic Medical Items

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

Adds a medical based tutorial aimed at describing every basic medical item a non-Corpsman Marine can access and use, and then describing where they come from and how to store them. 

The tutorial goes something like this.
- Tutorial loads, it begins to showcase every individual basic medical items, stating what they do, this includes bandages, splints, ointment, all kinds of autoinjectors and pill packets
- Afterwards, the tutorial will showcase the medical pouches, where to get them, and what they come with. It highlights that the injector pouch is unique whereas the other two pouches are free to get from squad vendors.
- The tutorial ends with letting the player access a generic prep vendor, encouraging the player to experiment with storing medical items, the player can then end the tutorial when they are ready. 

# Explain why it's good for the game

Educating new players on mechanics is good. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Tutorial describing basic medical items, their uses and where to get them has been added. 
/:cl:

### CODE REVIEW

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	if(!.)
+		return
+
+	init_mob()
+	message_to_player("This tutorial will discuss the medical tools you will use as a standard Marine.")
+	addtimer(CALLBACK(src, PROC_REF(intro_one)), 6 SECONDS)
+//pouch_one intro_one
+/datum/tutorial/marine/medical_basic_item/proc/intro_one()
+	message_to_player("You are capable of tending to your own basic injuries without needing Corpsman (sometimes called a Medic) to assist you.")
+	update_objective("")
```
Comment (Birdtalon) [inline code/datums/tutorial/marine/medical_basic_item.dm:25]: You only need to call this when you want to update the objective. No need to set it to a blank string every time.

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+	addtimer(CALLBACK(src, PROC_REF(kit_one)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/kit_one()
+	message_to_player("We will now show and discuss the basic medical items you can use. Afterwards, we will examine where you can find and store these items.")
+	update_objective("")
+	addtimer(CALLBACK(src, PROC_REF(kit_two)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/kit_two()
+	SIGNAL_HANDLER
```
Comment (Birdtalon) [inline code/datums/tutorial/marine/medical_basic_item.dm:39]: `SIGNAL_HANDLER` is only needed if you're calling this from a signal using `RegisterSignal()`. Not needed if you're just using a timer to call it.

File: code/game/machinery/vending/vendor_types/squad_prep/tutorial.dm [hunk 3]
```diff
@@ -28,3 +28,90 @@ GLOBAL_LIST_INIT(cm_vending_clothing_tutorial, list(
...
+/obj/structure/machinery/cm_vending/clothing/tutorialmed/get_listed_products(mob/user)
+	return GLOB.cm_vending_clothing_tutorialstandard
+
+//Standard Med Vendor 1
+GLOBAL_LIST_INIT(cm_vending_clothing_tutorial_medic, list(
+		list("POUCHES (CHOOSE 1)", 0, null, null, null),
+		list("First-Aid Pouch (Refillable Injectors)", 0, /obj/item/storage/pouch/firstaid/full, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
+	))
+
+/obj/structure/machinery/cm_vending/clothing/medic_equip
```
Comment (Birdtalon) [inline code/game/machinery/vending/vendor_types/squad_prep/tutorial.dm:68]: This typepath should probably explicitly say it's tutorial so someone doesn't use e by accident elsewhere.

**Review by Birdtalon — CHANGES_REQUESTED** [review]
Couple of things I've noticed from a brief scan. You also need to make sure you include your new .dm file in the .dme
I will try to have a proper look at this soon.

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+/datum/tutorial/marine/medical_basic_item/proc/intro_two()
+	message_to_player("The knowledge and ability to tend to your own injuries will allow you to not only survive, but thrive, on the battlefield.")
+	addtimer(CALLBACK(src, PROC_REF(kit_one)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/kit_one()
+	message_to_player("We will now show and discuss the basic medical items you can use. Afterwards, we will examine where you can find and store these items.")
+	addtimer(CALLBACK(src, PROC_REF(kit_two)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/kit_two()
+	message_to_player("These are <b>Bandages</b>. You can use these to heal <b>Brute</b> damage and stop bleeding. You always want to carry one of these.")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:36]: Bandages aren't proper nouns

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+	var/obj/item/stack/medical/bruise_pack/bandage = new(loc_from_corner(0, 5))
+	add_to_tracking_atoms(bandage)
+	add_highlight(bandage)
+	addtimer(CALLBACK(src, PROC_REF(kit_three)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/kit_three()
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/stack/medical/bruise_pack, bandage)
+	remove_highlight(bandage)
+	message_to_player("These are <b>Ointments</b>. You can use these to heal <b>Burn</b> damage. These are less important to carry, but are useful if you have the space for it.")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:45]: Ditto

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+	var/obj/item/stack/medical/ointment/cream = new(loc_from_corner(0, 6))
+	add_to_tracking_atoms(cream)
+	add_highlight(cream)
+	addtimer(CALLBACK(src, PROC_REF(kit_four)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/kit_four()
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/stack/medical/ointment, cream)
+	remove_highlight(cream)
+	message_to_player("These are <b>Splints</b>. You can use these to stabalize <b>fractures</b>. These are very important to use, as a unsplinted fracture can cause permanent injuries.")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:54]: Ditto

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+	SIGNAL_HANDLER
+
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/storage/pill_bottle/packet/bicaridine, packetbic)
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/storage/pill_bottle/packet/kelotane, packetkel)
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/storage/pill_bottle/packet/tramadol, packettram)
+	remove_highlight(packetbic)
+	remove_highlight(packetkel)
+	remove_highlight(packettram)
+	message_to_player("Take this moment to pause and review the previous chat logs, or the medical items on the floor. When you are ready, please take the food bar and eat it.")
+	update_objective("Eat the food bar when you are ready to continue")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:142]:
```suggestion
	update_objective("Eat the food bar when you are ready to continue.")
```

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+/datum/tutorial/marine/medical_basic_item/proc/kit_twelve()
+	SIGNAL_HANDLER
+
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/storage/pill_bottle/packet/bicaridine, packetbic)
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/storage/pill_bottle/packet/kelotane, packetkel)
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/item/storage/pill_bottle/packet/tramadol, packettram)
+	remove_highlight(packetbic)
+	remove_highlight(packetkel)
+	remove_highlight(packettram)
+	message_to_player("Take this moment to pause and review the previous chat logs, or the medical items on the floor. When you are ready, please take the food bar and eat it.")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:141]:
```suggestion
	message_to_player("Take this moment to pause and review the previous chat logs or the medical items on the floor. When you are ready, please take the food bar and eat it.")
```

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+	prep_vendor.req_access = list()
+	var/obj/item/reagent_container/food/snacks/protein_pack/food = new(loc_from_corner(3, 3))
+	add_to_tracking_atoms(food)
+	add_highlight(food)
+	RegisterSignal(tutorial_mob, COMSIG_MOB_EATEN_SNACK, PROC_REF(the_end))
+
+/datum/tutorial/marine/medical_basic_item/proc/the_end()
+	SIGNAL_HANDLER
+
+	message_to_player("Good luck Marine")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:264]:
```suggestion
	message_to_player("Good luck Marine.")
```

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+/datum/tutorial/marine/medical_basic_item/proc/pouch_ten()
+	message_to_player("It will be up to you to figure out which of these you should take. You can consider using these pouches, or placing some of them in your satchel/backpack.")
+	addtimer(CALLBACK(src, PROC_REF(pouch_eleven)), 6 SECONDS)
+
+/datum/tutorial/marine/medical_basic_item/proc/pouch_eleven()
+	SIGNAL_HANDLER
+
+	message_to_player("Feel free to examine these items, and use the prep vendor to experiment with storing these medical items. When you are finished, eat the food to end the tutoiral.")
+	update_objective("Eat the food bar when you are ready to end the tutorial")
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:252]:
```suggestion
	update_objective("Eat the food bar when you are ready to end the tutorial.")
```

File: code/datums/tutorial/marine/medical_basic_item.dm [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+	new /obj/structure/surface/table/almayer(loc_from_corner(4, 4))
+	new /obj/structure/surface/table/almayer(loc_from_corner(5, 4))
+	var/obj/structure/machinery/cm_vending/clothing/tutorialmed/clothing_vendor = new(loc_from_corner(5, 0))
+	add_to_tracking_atoms(clothing_vendor)
+	var/obj/structure/machinery/cm_vending/clothing/tutorial_medic_equip/medic_vendor_alpha = new(loc_from_corner(5, 1))
+	add_to_tracking_atoms(medic_vendor_alpha)
+	var/obj/structure/machinery/cm_vending/clothing/tutorial_medic_equip_bravo/medic_vendor_bravo = new(loc_from_corner(5, 2))
+	add_to_tracking_atoms(medic_vendor_bravo)
+	var/obj/structure/machinery/cm_vending/sorted/uniform_supply/tutorial_medic_equip_charlie/medic_vendor_charlie = new(loc_from_corner(5, 3))
+	add_to_tracking_atoms(medic_vendor_charlie)
```
Comment (Zonespace27) [inline code/datums/tutorial/marine/medical_basic_item.dm:292]: The names are confusing because the NATO alphabet is typically for squads.

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
requested changes made

**Review by Birdtalon — CHANGES_REQUESTED** [review]
Overall a good overview of standard medical and the code is good.

Although I personally thought that the pacing might be a little too fast for a new player when describing what the items are.

I also managed to break your tutorial somehow by softlocking myself at the stage where the player has to vend a pull packet pouch and there wasn't any to vend.
dreamseeker_2024-01-13_14-28-55

I would like to see a little work on the pacing though so it doesn't feel like information overload and we actually get the player to use more of the items and do stuff within the tutorial. Explain an item or two, then get the player to use them for themselves and see how they work- then move onto the next. The existing medical tutorial is a good example.  This will be more memorable to a player and will flow more easily into the next tutorial and so on and so on.

I'll let you have a think about this and mark your PR as ready for review when you're ready.

**Comment by Steelpoint** [comment]
I'll reopen once I've got free time to work out the PR and make changes, just closing to stop the auto-close

**Pull request closed without merging**